src/chainparamsbase.h

```diff
@@ -24,9 +26,12 @@ class CBaseChainParams
     const std::string& DataDir() const { return strDataDir; }
     int RPCPort() const { return nRPCPort; }
 
-protected:
-    CBaseChainParams() {}
+    CBaseChainParams(const std::string& strDataDir, int nRPCPort) {
```

Comment:
codestyle: just use data_dir and rpc_port.

Reply:
and then `nRPCPort(rpc_port), strDataDir(data_dir) {}` constructor initializer list.